src/EventListener/DataContainer/Theme.php

```diff
@@ -0,0 +1,379 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * Copyright (c) 2005-2018 Leo Feyer
+ *
+ * @license LGPL-3.0+
+ */
+
+namespace Contao\CoreBundle\EventListener\DataContainer;
+
+use Contao\Backend;
+use Contao\BackendUser;
+use Contao\CoreBundle\Exception\AccessDeniedException;
+use Contao\CoreBundle\Image\ImageFactoryInterface;
+use Contao\FilesModel;
+use Contao\Image;
+use Contao\StringUtil;
+use Contao\StyleSheets;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\HttpFoundation\Session\SessionInterface;
+use Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorageInterface;
+
+class Theme
+{
+    /** @var RequestStack */
+    protected $requestStack;
+
+    /** @var SessionInterface */
+    protected $session;
+
+    /** @var BackendUser */
+    protected $user;
+
+    /** @var ImageFactoryInterface */
+    private $imageFactory;
+
+    /**
+     * Theme constructor.
+     *
+     * @param RequestStack          $requestStack
+     * @param SessionInterface      $session
+     * @param TokenStorageInterface $tokenStorage
+     * @param ImageFactoryInterface $imageFactory
+     */
+    public function __construct(
+        RequestStack $requestStack,
+        SessionInterface $session,
+        TokenStorageInterface $tokenStorage,
+        ImageFactoryInterface $imageFactory
+    ) {
+        $this->requestStack = $requestStack;
+        $this->session      = $session;
+        $this->user         = $tokenStorage->getToken()->getUser();
+        $this->imageFactory = $imageFactory;
+    }
+
+    /**
+     * Check permissions to edit the table
+     *
+     * @throws AccessDeniedException
+     */
+    public function onCheckPermission(): void
+    {
+        if ($this->user->isAdmin) {
+            return;
+        }
+
+        $request = $this->requestStack->getCurrentRequest();
+
+        // Check the theme import and export permissions (see #5835)
+        switch ($request->get('key')) {
+            case 'importTheme':
+                if (!$this->user->hasAccess('theme_import', 'themes')) {
+                    throw new AccessDeniedException('Not enough permissions to import themes.');
+                }
+                break;
+
+            case 'exportTheme':
+                if (!$this->user->hasAccess('theme_import', 'themes')) {
+                    throw new AccessDeniedException('Not enough permissions to export themes.');
+                }
+                break;
+        }
+    }
+
+
+    /**
+     * Add an image to each record
+     *
+     * @param array  $row
+     * @param string $label
+     *
+     * @return string
+     */
+    public function onAddPreviewImage(array $row, string $label): string
+    {
+        if (!$row['screenshot']) {
+            return $label;
+        }
+
+        $objFile = FilesModel::findByUuid($row['screenshot']);
```

Comment:
In case you want to add unit tests, you need to be able to mock this. Inject `ContaoFrameworkInterface $framework` (you can also use setter injection because if possible we'd like to drop that one day): 
`$this->framework->getAdapter('Contao\FilesModel')->findByUuid($row['screenshot']);`.